Index: uinc/bits/uClibc_config.h
===================================================================
--- uinc/bits/uClibc_config.h	(revision 7124)
+++ uinc/bits/uClibc_config.h	(working copy)
@@ -104,7 +104,7 @@
 #undef __UCLIBC_HAS_CTYPE_CHECKED__
 #undef __UCLIBC_HAS_CTYPE_ENFORCED__
 #define __UCLIBC_HAS_WCHAR__ 1
-#undef __UCLIBC_HAS_LOCALE__
+#define __UCLIBC_HAS_LOCALE__ 1
 #undef __UCLIBC_HAS_GLIBC_CUSTOM_PRINTF__
 #undef __USE_OLD_VFPRINTF__
 #define __UCLIBC_PRINTF_SCANF_POSITIONAL_ARGS__ 9
Index: pkg/uclibc/libc/string/Makefrag
===================================================================
--- pkg/uclibc/libc/string/Makefrag	(revision 7124)
+++ pkg/uclibc/libc/string/Makefrag	(working copy)
@@ -13,7 +13,8 @@
 	wcscasecmp.c wcscat.c wcschrnul.c wcschr.c wcscmp.c wcscpy.c wcscspn.c \
 	wcsdup.c wcslen.c wcsncasecmp.c wcsncat.c wcsncmp.c wcsncpy.c \
 	wcsnlen.c wcspbrk.c wcsrchr.c wcsspn.c wcsstr.c wcstok.c wmemchr.c \
-	wmemcmp.c wmemcpy.c wmemmove.c wmempcpy.c wmemset.c wcpcpy.c wcpncpy.c
+	wmemcmp.c wmemcpy.c wmemmove.c wmempcpy.c wmemset.c wcpcpy.c wcpncpy.c \
+	wcsxfrm.c wcslcpy.c strxfrm.c
 
 LIBC_SRCFILES += $(patsubst %.c, pkg/uclibc/libc/string/%.c, $(STRING_CSRC))
 
Index: pkg/uclibc/libc/misc/Makefrag
===================================================================
--- pkg/uclibc/libc/misc/Makefrag	(revision 7124)
+++ pkg/uclibc/libc/misc/Makefrag	(working copy)
@@ -44,7 +44,8 @@
         __ctype_assert.c isctype.c
 
 MISC_LOCALE_CSRC := \
-	setlocale.c localeconv.c _locale_init.c nl_langinfo.c
+	setlocale.c localeconv.c _locale_init.c nl_langinfo.c \
+	newlocale.c __locale_mbrtowc_l.c locale_data.c
 
 MISC_TIME_CSRC := \
 	asctime.c asctime_r.c clock.c ctime.c ctime_r.c gmtime.c gmtime_r.c \
