--- dhcp-3.0.5_old/Makefile.conf	2005-06-16 12:39:35.000000000 -0700
+++ dhcp-3.0.5/Makefile.conf	2007-06-12 13:18:00.000000000 -0700
@@ -78,6 +78,11 @@
 # removes the comment characters from the appropriate set of
 # assignments, and writes the output to Makefile.
 
+##--jos64--
+CC=acpkg-wrap-gcc
+CF=cf/jos64.h
+##--jos64--
+
 ## AIX 4.3
 ##--aix--
 #CF = cf/aix.h
--- dhcp-3.0.5_old/includes/osdep.h	2005-09-28 11:58:27.000000000 -0700
+++ dhcp-3.0.5/includes/osdep.h	2007-06-12 13:26:41.000000000 -0700
@@ -133,6 +133,10 @@
 # endif
 #endif
 
+#ifdef JOS_USER
+# include "cf/jos64.h"
+#endif
+
 /* snprintf/vsnprintf hacks.  for systems with no libc versions only. */
 #ifdef NO_SNPRINTF
   extern int isc_print_snprintf(char *, size_t, const char *, ...);
--- dhcp-3.0.5_old/configure	2004-09-10 14:02:30.000000000 -0700
+++ dhcp-3.0.5/configure	2007-05-23 00:03:38.000000000 -0700
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-#sysname=$1
+sysname=jos64
 
 while [ $# != 0 ]; do
   if [ x$1 = x--with-nsupdate ]; then
--- dhcp-3.0.5_old/includes/cf/jos64.h	1969-12-31 16:00:00.000000000 -0800
+++ dhcp-3.0.5/includes/cf/jos64.h	2007-06-12 14:57:55.000000000 -0700
@@ -0,0 +1,8 @@
+
+
+#define LINUX_SLASHPROC_DISCOVERY
+#define PROCDEV_DEVICE "/proc/net/dev"
+
+
+#define HAVE_SO_BINDTODEVICE
+#define HAVE_SIOCGIFHWADDR
