--- dmalloc-5.4.2_bad/configure	2004-10-19 07:51:21.000000000 -0700
+++ dmalloc-5.4.2/configure	2006-04-02 00:15:38.000000000 -0800
@@ -4573,7 +4573,7 @@
 
 echo "$as_me:$LINENO: checking basic-block size" >&5
 echo $ECHO_N "checking basic-block size... $ECHO_C" >&6
-ac_cv_page_size=0
+ac_cv_page_size=12
 if test $ac_cv_page_size = 0; then
    if test "$cross_compiling" = yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling" >&5
