--- gcc-3.4.4_bad/gcc/Makefile.in	2005-02-24 01:26:57.000000000 -0800
+++ gcc-3.4.4/gcc/Makefile.in	2006-03-20 05:05:20.000000000 -0800
@@ -606,7 +606,9 @@
 
 # Native linker and preprocessor flags.  For x-fragment overrides.
 BUILD_LDFLAGS=$(LDFLAGS)
-BUILD_CPPFLAGS=$(ALL_CPPFLAGS)
+BUILD_CPPFLAGS=
+# Don't want to use includes and libs intended for host and target system
+#BUILD_CPPFLAGS=$(ALL_CPPFLAGS)
 
 # Actual name to use when installing a native compiler.
 GCC_INSTALL_NAME := $(shell echo gcc|sed '$(program_transform_name)')
