--- a/configure.ac
+++ b/configure.ac
@@ -18,6 +18,12 @@ AC_CONFIG_HEADER([include/X11/XlibConf.h])
 AC_PROG_LIBTOOL
 AC_PROG_CC

+if test x"$cross_compiling" = xyes; then
+       AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+else
+       CC_FOR_BUILD="$CC"
+fi
+
 XORG_PROG_RAWCPP

 # Build with XCB support?
--- a/src/util/Makefile.in	2007-12-04 17:06:47.000000000 -0800
+++ b/src/util/Makefile.in	2007-12-04 17:07:28.000000000 -0800
@@ -80,7 +80,7 @@
 AWK = @AWK@
 BIGFONT_CFLAGS = @BIGFONT_CFLAGS@
 BIGFONT_LIBS = @BIGFONT_LIBS@
-CC = @CC@
+CC = $(CC_FOR_BUILD)
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
 CPP = @CPP@
