--- mairix-0.21/dfasyn/Makefile	2008/01/22 08:33:36	1.1
+++ mairix-0.21/dfasyn/Makefile	2008/01/22 08:33:59
@@ -15,10 +15,10 @@
 # 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA
 #
 
-CC=gcc
+override CC=gcc
 #CFLAGS=-g -Wall
 #CFLAGS=-O2 -pg
-CFLAGS=-Wall
+override CFLAGS=-Wall
 prefix?=/usr/local
 bindir=$(prefix)/bin
 mandir?=$(prefix)/man
--- mairix-0.21/writer.c	2008/01/28 06:43:03	1.1
+++ mairix-0.21/writer.c	2008/01/28 06:43:08
@@ -602,10 +602,6 @@
     report_error("munmap", filename);
     unlock_and_exit(2);
   }
-  if (fsync(fd) < 0) {
-    report_error("fsync", filename);
-    unlock_and_exit(2);
-  }
   if (close(fd) < 0) {
     report_error("close", filename);
     unlock_and_exit(2);
