--- perl-5.8.8_bad/uconfig.h	2005-10-12 05:03:14.000000000 -0700
+++ perl-5.8.8/uconfig.h	2007-02-04 15:26:02.000000000 -0800
@@ -110,7 +110,7 @@
  *	This symbol, if defined, indicates that the dup2 routine is
  *	available to duplicate file descriptors.
  */
-/*#define HAS_DUP2	/ **/
+#define HAS_DUP2	/ **/
 
 /* HAS_FCHMOD:
  *	This symbol, if defined, indicates that the fchmod routine is available
