--- sfslite-0.8.9pre3_bad/acsfs.m4	2006-07-24 17:14:30.000000000 -0700
+++ sfslite-0.8.9pre3/acsfs.m4	2006-11-01 14:22:18.000000000 -0800
@@ -319,7 +319,8 @@
   return n != xn + 1;
 }
 changequote([,])],
-	sfs_cv_egid_in_grouplist=yes, sfs_cv_egid_in_grouplist=no))
+	sfs_cv_egid_in_grouplist=yes, sfs_cv_egid_in_grouplist=no, 
+	sfs_cv_egid_in_grouplist=no))
 if test $sfs_cv_egid_in_grouplist = yes; then
 	AC_DEFINE(HAVE_EGID_IN_GROUPLIST, 1,
 	  Define if the first element of a grouplist is the effective gid)
--- sfslite-0.8.9pre3_bad/async/arandom.c	2006-05-31 08:02:38.000000000 -0700
+++ sfslite-0.8.9pre3/async/arandom.c	2006-11-01 14:27:16.000000000 -0800
@@ -21,6 +21,8 @@
  *
  */
 
+extern unsigned int arc4random(void);
+
 #include "sysconf.h"
 
 /* XXX - need initializer because of broken Apple Darwin linker */
--- sfslite-0.8.9pre3_bad/async/dftables.c	2006-05-31 08:02:37.000000000 -0700
+++ sfslite-0.8.9pre3/async/dftables.c	2006-11-01 14:32:12.000000000 -0800
@@ -44,6 +44,7 @@
 #include <stdio.h>
 #include <string.h>
 
+#define _ASYNC_SYSCONF_H_ 1
 #include "internal.h"
 
 #define DFTABLES          /* maketables.c notices this */
--- sfslite-0.8.9pre3_bad/async/Makefile.am	2006-07-24 17:14:29.000000000 -0700
+++ sfslite-0.8.9pre3/async/Makefile.am	2006-11-01 14:40:56.000000000 -0800
@@ -13,6 +13,8 @@
 noinst_PROGRAMS = dftables
 sfsexec_PROGRAMS = aiod mmcd
 
+aiod_LINK = $(LINK)
+
 libasync_la_SOURCES = \
 arandom.c clock_gettime.c flock.c rwfd.c stktrace.c suidprotect.c \
 maketables.c pcre.c study.c \
@@ -51,6 +53,9 @@
 dftables_SOURCES = dftables.c
 dftables_LDADD =
 
+dftables: dftables.c
+	gcc $< -o $@
+
 pcre.o pcre.lo: chartables.c
 chartables.c: dftables
 	./dftables $@
--- sfslite-0.8.9pre3_bad/async/spawn.C	2006-05-31 08:02:37.000000000 -0700
+++ sfslite-0.8.9pre3/async/spawn.C	2006-11-01 14:36:05.000000000 -0800
@@ -72,7 +72,7 @@
 
 #ifdef MAINTAINER
   if (afork_debug) {
-    warn ("AFORK_DEBUG: child process pid %d\n", getpid ());
+      warn ("AFORK_DEBUG: child process pid %d\n", (int)getpid ());
     sleep (7);
   }
 #endif /* MAINTAINER */
--- sfslite-0.8.9pre3_bad/async/mmcd.c	2006-07-10 07:38:21.000000000 -0700
+++ sfslite-0.8.9pre3/async/mmcd.c	2006-11-01 14:42:10.000000000 -0800
@@ -167,7 +167,7 @@
   }
 
   fprintf (stderr, "%s: starting up; file=%s; pid=%d\n", 
-	   progname, mmap_file, getpid ());
+	   progname, mmap_file, (int)getpid ());
 
   set_signal_handlers ();
   
--- sfslite-0.8.9pre3_bad/Makefile.am	2006-07-17 13:22:22.000000000 -0700
+++ sfslite-0.8.9pre3/Makefile.am	2006-11-01 14:53:57.000000000 -0800
@@ -24,8 +24,8 @@
 ARPCGEN =
 endif
 
-SUBDIRS = async rpcc arpc crypt $(ARPCGEN) $(LIBSFS) $(SVC) sfsmisc contrib \
-	tame libtame $(TUTORIAL) 
+SUBDIRS = async arpc crypt $(ARPCGEN) $(LIBSFS) $(SVC) sfsmisc contrib \
+	  libtame $(TUTORIAL) 
 DIST_SUBDIRS = $(SUBDIRS)
 #
 # end sfslite change
--- sfslite-0.8.9pre3_bad/configure.in	2006-07-24 17:14:30.000000000 -0700
+++ sfslite-0.8.9pre3/configure.in	2006-11-01 14:54:13.000000000 -0800
@@ -446,8 +446,8 @@
 SFS_LIBSFS
 SFS_CALLBACK
 
-AC_OUTPUT(Makefile async/Makefile rpcc/Makefile arpc/Makefile
-	crypt/Makefile sfsmisc/Makefile contrib/Makefile tame/Makefile
+AC_OUTPUT(Makefile async/Makefile arpc/Makefile
+	crypt/Makefile sfsmisc/Makefile contrib/Makefile
 	svc/Makefile libtame/Makefile arpcgen/Makefile libsfs/Makefile)
 
 if test "$with_tutorial" = "yes"; then
--- sfslite-0.8.9pre3_bad/arpc/authopaque.C	2006-05-31 08:02:07.000000000 -0700
+++ sfslite-0.8.9pre3/arpc/authopaque.C	2006-11-01 14:50:27.000000000 -0800
@@ -24,10 +24,12 @@
 
 #include "arpc.h"
 
+#if 0
 extern "C" {
   /* for SunOS 5.7 */
   bool_t xdr_opaque_auth(XDR *, struct opaque_auth *);
 }
+#endif
 
 static int authopaque_marshal (AUTH *, XDR *);
 static void authopaque_destroy (AUTH *);
--- sfslite-0.8.9pre3_bad/crypt/Makefile.am	2006-05-31 08:02:26.000000000 -0700
+++ sfslite-0.8.9pre3/crypt/Makefile.am	2006-11-01 14:52:10.000000000 -0800
@@ -50,6 +50,12 @@
 #tst_DEPENDENCIES = 
 #endif
 
+LINK = $(LIBTOOL) --tag=CC --mode=link $(CC) $(AM_CFLAGS) $(CFLAGS) \
+		$(AM_LDFLAGS) $(LDFLAGS) -o $@
+
+tst_LINK = $(LINK)
+timing_LINK = $(LINK)
+
 if REPO
 sfscrypt_repo_OBJECTS = $(libsfscrypt_la_OBJECTS) \
 	$(LIBARPC) $(LIBASYNC)
